### PR TITLE
Initial CMake and Linux support

### PR DESCRIPTION
**Pull request by MrAnno**
This PR aims to introduce Linux support by using CMake.

```
XPanel [6872] [INFO]:plugin start
XPanel [6872] [INFO]:XPanel ver 1.3
XPanel [6872] [INFO]:Built at Oct  8 2022 01:33:52
Loaded: /home/anno/.local/share/Steam/steamapps/common/X-Plane 11/Resources/plugins/XPanel/64/lin.xpl (xpanel).

XPanel [6872] [INFO]:plugin enable called
...
```

The new CMake build system compiles Linux binaries for XPanel, and will eventually be able to produce a macOS plugin and generate the Visual Studio project, which is currently used as the main build solution.

Before releasing a new version, the version needs to be updated in the main `CMakeLists.txt` file:
```
project(XPanel VERSION 1.3 LANGUAGES CXX)
```

Attaching Linux binaries to the release is currently not possible, as the Linux plugin depends on the system's Lua and hidapi libraries.

I've added install targets as well, the install directory on Linux looks like this:
```
.
├── doc
│   ├── configuration.md
│   ├── lua.md
│   └── xpanel.mdj
├── sample_configs
│   ├── 747-200
│   ├── airfoillabs c172sp
│   ├── ...
└── XPanel
    └── 64
        └── lin.xpl
```

TODO:
- [x] Windows support with CMake (generate the VS project)
- [ ] Run tests on Linux
- [x] Manual test with proper hardware on Linux (I'm gonna need some help :) )
- [x] Linux build in CI

### CODE REVIEW

**Comment by norberttak** [comment]
Thanks @MrAnno to work on this PR. I'm really excited to see these changes :-)
Regarding the test HW: I have a spare Arduino Leonardo board. I'll connect some switches and LEDs to it, so you can use it to test the plugin on Linux.

**Comment by MrAnno** [comment]
@norberttak Thank you :)

I've added CMake-based Windows build possibilities as well.
You can either open XPanel as a CMake project in Visual Studio or you can generate VS solution/project files using CMake.

My plan was to remove the VS project files once I'm done so that you don't need to maintain 2 different build systems.
It turned out that both CMake-based build options give a strange feeling, so I think it would be worth keeping the native VS project for a little while longer.

Here is a screenshot of how all of this looks like in VS:

image

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+include(CTest)
+
+add_subdirectory(hidapi)
+add_subdirectory(LUA)
+add_subdirectory(SDK)
+
+add_subdirectory(src)
+add_subdirectory(test)
+
```
Comment (norberttak) [inline CMakeLists.txt:26]: on Windows it is more convenient to copy the lua and hidapi dll to the output folder.

on all plataform the board-config.ini is also needed if you use an own HW (like an Arduino board)

I'm not very familiar with CMake's syntax but something like that could help:

install(FILES config/board-config.ini DESTINATION XPanel/64)
if (WIN32)
install(FILES lua/lua54.dll DESTINATION XPanel/64)
install(FILES hidapi/lib/hidapi.dll DESTINATION XPanel/64)
endif()

Comment (MrAnno) [reply CMakeLists.txt:27]: Fixed. Thanks.

**Comment by norberttak** [comment]
Thanks @MrAnno for this PR, I really like it :-)

**Pull request merged**